encord/client.py

```diff
@@ -816,16 +817,16 @@ def model_delete(self, uid: str) -> bool:
 
     def model_inference(
         self,
-        uid,
-        file_paths=None,
-        base64_strings=None,
-        conf_thresh=0.6,
-        iou_thresh=0.3,
-        device="cuda",
-        detection_frame_range=None,
-        allocation_enabled=False,
-        data_hashes=None,
-        rdp_thresh=0.005,
+        uid: str,
+        file_paths: Optional[List[str]] = None,
+        base64_strings: Optional[List[str]] = None,
+        conf_thresh: float = 0.6,
+        iou_thresh: float = 0.3,
+        device: Union[str, Devices] = Devices.CUDA,
+        detection_frame_range: Optional[List[int]] = None,
+        allocation_enabled: bool = False,
+        data_hashes: Optional[List[str]] = None,
+        rdp_thresh: float = 0.005,
     ):
         """
         This function is documented in :meth:`encord.project.Project.model_inference`.
```

Comment:
That's a great change. Maybe in the docstring of the other function you can explain briefly what combination of inputs one can choose. 
Not all users of the SDK will read the implementation, and maybe we can give them a heads up about what arguments are Optional in what circumstances.

Reply:
Yes that's a good idea. It would be nice if that was reflected in the type hints, but I'm not sure if that's possible in python